src/coverlet.msbuild.tasks/InstrumentationTask.cs

```diff
@@ -106,7 +106,20 @@ public override bool Execute()
                 var excludedSourceFiles = _excludeByFile?.Split(',');
                 var excludeAttributes = _excludeByAttribute?.Split(',');
 
-                Coverage coverage = new Coverage(_path, includeFilters, includeDirectories, excludeFilters, excludedSourceFiles, excludeAttributes, _includeTestAssembly, _singleHit, _mergeWith, _useSourceLink, _logger, (IInstrumentationHelper)DependencyInjection.Current.GetService(typeof(IInstrumentationHelper)));
+                Coverage coverage = new Coverage(_path,
+                    includeFilters,
+                    includeDirectories,
+                    excludeFilters,
+                    excludedSourceFiles,
+                    excludeAttributes,
+                    _includeTestAssembly,
+                    _singleHit,
+                    _mergeWith,
+                    _useSourceLink,
+                    _logger,
+                    (IFileSystem) DependencyInjection.Current.GetService(typeof(IFileSystem)),
+                    (IInstrumentationHelper) DependencyInjection.Current.GetService(typeof(IInstrumentationHelper)));
+
                 CoveragePrepareResult prepareResult = coverage.PrepareModules();
                 InstrumenterState = new TaskItem(System.IO.Path.GetTempFileName());
                 using (var instrumentedStateFile = new FileStream(InstrumenterState.ItemSpec, FileMode.Open, FileAccess.Write))
```

Comment:
also for new FileStream(InstrumenterState.ItemSpec, FileMode.Open) and surround by using please(my past error)

Reply:
Done